packages/react-instantsearch-core/src/core/indexUtils.js

```diff
@@ -8,13 +8,20 @@ export function getIndexId(context) {
 }
 
 export function getResults(searchResults, context) {
-  if (searchResults.results && !searchResults.results.hits) {
-    return searchResults.results[getIndexId(context)]
-      ? searchResults.results[getIndexId(context)]
-      : null;
-  } else {
-    return searchResults.results ? searchResults.results : null;
+  if (!searchResults.results) {
+    return null;
+  }
+
+  if (searchResults.results.hits) {
+    return searchResults.results;
   }
+
+  const indexId = getIndexId(context);
+  if (searchResults.results[indexId]) {
+    return searchResults.results[indexId];
+  }
+
+  return null;
```

Comment:
We could avoid the double `null` return with the condition `searchResults.results` inline: 

```js
export function getResults(searchResults, context) {
  if (searchResults.results && searchResults.results.hits) {
    return searchResults.results;
  }

  const indexId = getIndexId(context);
  if (searchResults.results && searchResults.results[indexId]) {
    return searchResults.results[indexId];
  }

  return null;
}
```

IMO it's easier to read but I let you chose which one you want to implement.

Reply:
yes, thought of that as an alternative, went for this style because it's less comparisons

Reply:
```js
export function getResults(searchResults, context) {
  if (searchResults.results) {
    if (searchResults.results.hits) {
      return searchResults.results;
    }

    const indexId = getIndexId(context);
    if (searchResults.results[indexId]) {
      return searchResults.results[indexId];
    }
  }

  return null;
}
```

this is also somewhat easier since it has a return less, but all three styles suggested are fine I think

Reply:
@Haroenv Take one or the other suggestion is clearer that what we have right now.